message SingleReadSet {
    required string filename = 1;
    optional double mismatch_prob = 2 [default = 0.01];
    optional double min_prob_start = 3 [default = -10];
    optional double min_prob_per_base = 4 [default = -0.7];
    optional double penalty_constant = 5 [default = 0];
    optional int32 penalty_step = 6 [default = 0];
    optional double weight = 7 [default = 1];
}

message Config {
    required string starting_graph = 1;
    optional string output_file = 3 [default = 'output.fasta'];

    // Annealing settings: Temperature at iteration n is t0 / (log(n/n_divisor + 1))
    optional double t0 = 4 [default = 0.08];
    optional double n_divisor = 5 [default = 1];

    optional int32 num_iterations = 6 [default = 100];

    repeated SingleReadSet single_short_reads = 2;
}
